tests/osn-tests/src/test_osn_global.ts

```diff
@@ -50,10 +57,92 @@ describe('osn_global', () => {
             try {
                 source = osn.Global.getOutputSource(1);
             } catch(e) {
-                throw new Error("failed to set source to output channel");
+                throw new Error(getCppErrorMsg(e));
             }
             
             expect(source.name).to.equal('test_osn_global_source');
         });
+
+        it('should fail if trying to get from empty output channel', () => {
+            let source: ISource;
+
+            try {
+                source = osn.Global.getOutputSource(5);
+            } catch(e) {
+                throw new Error(getCppErrorMsg(e));
+            }
+
+            expect(source).to.equal(undefined);
+        });
+    });
+
+    context('# GetOutputFlagsFromId', () => {
+        it('should get flags (capabilities) of a source type', () => {
+            let sourceType: string;
+            let sourceTypes: string[];
+            let flags: number = 0;
+
+            try {
+                sourceTypes = osn.InputFactory.types();
+            } catch(e) {
+                throw new Error(getCppErrorMsg(e));
+            }
+
+            expect(sourceTypes.length).to.not.equal(0);
+
+            for (sourceType of sourceTypes)
+            {
+                try {
+                    flags = osn.Global.getOutputFlagsFromId(sourceType);
+                } catch(e) {
+                    throw new Error(getCppErrorMsg(e));
+                }
+
+                expect(flags).to.not.equal(0);
+                flags = 0;
+            }
+        });
+    });
+
+    context('# LaggedFrames', () => {
+        it('should get lagged frames value', () => {
+            let laggedFrames: number;
+
+            expect(function() {
+                laggedFrames = osn.Global.laggedFrames;
+            }).to.not.throw();
+        });
+    });
+
+    context('# TotalFrames', () => {
+        it('should get total frames value', () => {
+            let totalFrames: number = undefined;
+
+            expect(function() {
+                totalFrames = osn.Global.totalFrames;
+            }).to.not.throw();
+
+            expect(totalFrames).to.not.equal(undefined);
+        });
+    });
+
+    context('# SetLocale', () => {
+        it('shoud set locale', () => {
+            expect(function() {
+                osn.Global.locale = 'pt-BR';
+            }).to.not.throw();
+        });
+    });
+
+    context('# GetLocale', () => {
+        it('shoud get locale', () => {
+            let locale: string;
+
+            expect(function() {
+                locale = osn.Global.locale;
+            }).to.not.throw();
+
+            expect(locale).to.equal('pt-BR');
```

Comment:
I believe this is true on your machine, not on others. Let's set English so that it passes for everyone + Appveyor.

Reply:
The OBS process is indeed started with en-US as locale. Here I'm just testing if the function GetLocale is returning the value as expected. In the previous test case I used SetLocale to change it to 'pt-BR'.

Reply:
If some tests are depending from others we should make sure it is done in an explicit way. Otherwise it's gonna be very hard for others to maintain these tests and add others.

Reply:
Changed the SetLocale and GetLocale to be in the same test case. Now it is more clear what is happening and I'll pair Set and Get functions together from now on. It makes more sense to have them together.